extensions/amp-auto-ads/amp-auto-ads-spec.md

```diff
@@ -0,0 +1,268 @@
+<!---
+Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# <a name="amp-auto-ads-spec"></a> `amp-auto-ads` spec
+
+## Description
+`amp-auto-ads` uses a remotely served configuration to dynamically inject ads
+into the page.
+
+## Ad Placement Strategy
+Given a sufficient number of valid placements (supplied in the configuration),
+`amp-auto-ads` will try to insert additional ads within the following
+constraints:
+<ul>
+  <li>No more than 3 ads on the page (including any existing ads)</li>
+  <li>No injected ad within 500px (measured vertically) of another ad</li>
+  <li>
+    Ad can be injected without causing unacceptable re-flow (as determined by
```

Comment:
Rewrite as:

The injected ad does not cause any unacceptable re-flow (as determined by `attemptChangeSize`).